### PR TITLE
Fix flex children sections are showing on wrong elements

### LINKED ISSUE #685: Flex and Flex children sections are showing on wrong elements
When parent is set on display: flex, we show the flex section and flex child section on it, though, flex child should be only shown for it's children and flex section should not be shown on the child unless it's also set to display flex.

### PR DESCRIPTION
**Pull request by istarkov**
## Description

1. What is this PR about (link the issue and add a short description)

Fix flex children sections are showing on wrong elements
closes #685
most used functions are imported from style-info and all has tests there.

## Steps for reproduction

Create a few Boxes, and see that the Flex Child section is not shown.
Set `display: flex` for some Box. See that Flex Child section is not shown on the element. See that Flex Child section is shown on block children.

## Code Review

- [ ] hi @kof , @TrySound I want you to look namings, too much we have with styles Style, StyleDecl, StyleInfo etc.
I used useInstanceStyleData instead of useInstanceStyle because we already have similar useInstanceStyles

- [ ] hi @TrySound , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  if (rootInstance === undefined) {
+    return undefined;
+  }
+
+  const ancestorsAndSelf = utils.tree.getInstancePath(rootInstance, instanceId);
+
+  if (ancestorsAndSelf.length < 2) {
+    return undefined;
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts:104]:
```suggestion
    return;
```

File: apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        styleData[property] = value;
+      }
+    }
+    return styleData;
+  }, [selftAndCascadeInfo, inheritedInfo]);
+
+  return styleData;
+};
+
+export const useParentInstanceId = (instanceId: Instance["id"] | undefined) => {
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts:90]: getParentInstance should be probably a utility in shared/tree-utils with a test

Comment (kof) [reply apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts:90]: reason is this is a use case that is popular and we can optimize it later using a data structure

File: apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import { useStore } from "@nanostores/react";
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts:1]: sounds like file could be just instance-style-data.ts

Comment (kof) [reply apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts:1]: we already have a file called style-info.ts, seems like this logic is of the same type and should go in there as well

File: apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts (hunk 2 above)
Comment (TrySound) [reply apps/builder/app/builder/features/style-panel/shared/style-instance-style-data.ts:90]: @istarkov You can use instances index today. Test free.

```
const { parentInstanceById } = useStore(instancesIndex)
```

File: apps/builder/app/builder/features/style-panel/shared/style-info.ts [hunk 4]
```diff
@@ -282,3 +282,69 @@ export const useStyleInfo = () => {
...
+  const [breakpoints] = useBreakpoints();
+  const selectedBreakpoint = useStore(selectedBreakpointStore);
+  const selectedBreakpointId = selectedBreakpoint?.id;
+
+  const cascadedBreakpointIds = useMemo(
+    () => getCascadedBreakpointIds(breakpoints, selectedBreakpointId),
+    [breakpoints, selectedBreakpointId]
+  );
+
+  const selftAndCascadeInfo = useMemo(() => {
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/shared/style-info.ts:300]: typo

**Review by kof — APPROVED** [review]
looking good

**Pull request merged**